command/validator/registration/register_validator.go

```diff
@@ -79,6 +93,23 @@ func runCommand(cmd *cobra.Command, _ []string) error {
 		return err
 	}
 
+	if params.amountValue.Cmp(big.NewInt(0)) > 0 {
+		approveTxn, err := bridgeHelper.CreateApproveERC20Txn(params.amountValue,
+			contracts.StakeManagerContract, contracts.NativeERC20TokenContract, true)
+		if err != nil {
+			return err
+		}
+
+		receipt, err := txRelayer.SendTransaction(approveTxn, validatorAccount.Ecdsa)
+		if err != nil {
+			return err
+		}
+
+		if receipt.Status == uint64(types.ReceiptFailed) {
+			return fmt.Errorf("approve transaction failed on block %d", receipt.BlockNumber)
+		}
+	}
+
 	newValidatorAccount, err := wallet.NewAccountFromSecret(secretsManager)
```

Comment:
Do we need this if we created: `validatorAccount, err := validatorHelper.GetAccount(params.accountDir, params.accountConfig)`?